File: src/main/java/com/feiliks/rms/repositories/UserRepository.java
package com.feiliks.rms.repositories;

import com.feiliks.rms.entities.User;
import org.springframework.data.repository.CrudRepository;

public interface UserRepository extends CrudRepository<User, Long> {

    User findByUsername(String username);

    boolean existsByUsername(String username);
}

File: src/main/kotlin/com/feiliks/blog/repositories/UserRepository.kt
package com.feiliks.blog.repositories

import com.feiliks.blog.entities.UserEntity
import org.springframework.data.repository.CrudRepository

interface UserRepository : CrudRepository<UserEntity, Long> {

    fun findByUsername(username: String?): UserEntity?

    fun existsByUsername(username: String?): Boolean

}